docs/developer-docs/ai/inference.mdx

```diff
@@ -0,0 +1,63 @@
+
+# Decentralized AI Inference
+
+<MarkdownChipRow labels={["Intermediate", "Concept", "DeAI" ]} />
+
+It's possible for canister smart contracts to run inference in a number of ways, depending on the decentralization and performance requirements.
+
+## 1. Inference on-chain
+
+Currently, ICP supports on-chain inference of small models using AI libraries such as [Sonos Tract](https://github.com/sonos/tract) that compile to WebAssembly.
+Check out the [image classification example](/docs/current/developer-docs/ai/ai-on-chain) to learn how it works.
+The long-term [vision of DeAI on ICP](https://internetcomputer.org/roadmap#Decentralized%20AI-start) is to support on-chain GPU compute to enable both training and inference of larger models.
+
+### Examples
+
+- [GPT2](https://github.com/modclub-app/rust-connect-py-ai-to-ic/tree/main/internet_computer/examples/gpt2): An example of GPT2 running on-chain. `Rust`
+- [ELNA AI](https://github.com/elna-ai): A fully on-chain AI agent platform and marketplace. Supports both on-chain and off-chain LLMs. [Try it here](https://dapp.elna.ai/).
+- [Tensorflow on ICP](https://github.com/carlosarturoceron/decentAI): An Azle example that uses a pre-trained model for making predictions. `TypeScript`
+- [ICGPT](https://github.com/icppWorld/icgpt): A React frontend that uses a C/C++ backend running an LLM fully on-chain. [Try it here](https://icgpt.icpp.world/). `TypeScript` `C++`
+- [ArcMind AI](https://github.com/arcmindai/arcmindai): An autonomous agent using Chain of Thoughts for reasoning and actions. [Try it here](https://arcmindai.app). `Rust`
+
+### On-chain inference frameworks
+
+- [Sonos Tract](https://github.com/sonos/tract): An open-source AI inference engine written in Rust that supports ONNX, TensorFlow, PyTorch models and compiles to WebAssembly. `Rust`
+- [MotokoLearn](https://github.com/ildefons/motokolearn): A Motoko package that enables on-chain machine learning.
+  [The image classification example](https://github.com/dfinity/examples/tree/master/rust/image-classification) explains how to integrate it into a canister to run on ICP.
+  `Motoko`
+- [Rust-Connect-Py-AI-to-IC](https://github.com/jeshli/rust-connect-py-ai-to-ic): Open-source tool for deploying and running Python AI models on-chain using Sonos Tract.
+  `Rust`
+- [Burn](https://github.com/tracel-ai/burn): An open-source deep learning framework written in Rust that supports ONNX, PyTorch models and compiles to WebAssembly.
+  [The MNIST example](https://github.com/smallstepman/ic-mnist) explains how to integrate it into a canister to run on ICP. [Try it here](https://jsi2g-jyaaa-aaaam-abnia-cai.icp0.io/).
+  `Rust`
+- [Candle](https://github.com/huggingface/candle): a minimalist ML framework for Rust that compiles to WebAssembly.
+  [An AI chatbot example](https://github.com/ldclabs/ic-panda/tree/main/src/ic_panda_ai) shows how to run a Qwen 0.5B model in a canister on ICP.
+  `Rust`
+
+----
```

Comment:
```suggestion

```

Reply:
It would be useful for me to understand why a change is being suggested :) In this case, I saw the divider as something that visually helped with the organization of the page.